tests/gold_tests/pluginTest/lua/lua_states_stats.test.py

```diff
@@ -67,36 +71,27 @@
 ps.Env = ts.Env
 ps.ReturnCode = 0
 ps.Streams.stdout.Content = Testers.ContainsExpression("proxy.config.plugin.lua.max_states: 4", "expected 4 states")
-tr.TimeOut = 5
 tr.StillRunningAfter = ts
 
 # 1 Test - Exercise lua script
 tr = Test.AddTestRun("Lua hello")
 ps = tr.Processes.Default  # alias
 ps.Command = curl_and_args + ' http://hello/hello'
-ps.TimeOut = 5
 ps.ReturnCode = 0
 ps.Streams.stderr.Content = Testers.ContainsExpression("Hello, World", "hello world content")
-tr.TimeOut = 5
 tr.StillRunningAfter = ts
 
 # 2 Test - Check for metrics
 tr = Test.AddTestRun("Check for metrics")
-tr.DelayStart = 15  # 5s lag on metrics to update
-tr.TimeOut = 5
 ps = tr.Processes.Default  # alias
-ps.Env = ts.Env
-ps.Command = "traffic_ctl metric match lua"
+ps.Command = "bash -c '" + Test.TestDirectory + "/metrics.sh " + Test.TestDirectory + "'"
 ps.Env = ts.Env
 ps.ReturnCode = 0
-ps.Streams.stdout = "gold/metrics.stdout.gold"
 tr.StillRunningAfter = ts
 
 # 3 Test - Check for developer lifecycle stats
 tr = Test.AddTestRun("Check for lifecycle stats")
 ps = tr.Processes.Default  # alias
-ps.Command = "traffic_ctl plugin msg ts_lua print_stats"
+ps.Command = "bash -c '" + Test.TestDirectory + "/lifecycle_stats.sh " + Test.TestDirectory + "'"
```

Comment:
Same here..  One should almost never need to call Test.TestDirectory.

Just add someplace a `Test.Copy('lifecycle_stats.sh")`